BreadPlayer.Views.UWP/Core/SharedLogic.cs

```diff
@@ -47,7 +51,26 @@ public SharedLogic()
             InitializeCore.IsMobile = ApiInformation.IsApiContractPresent("Windows.Phone.PhoneContract", 1);
 
             InitializeCore.IsMobile = Window.Current?.Bounds.Width <= 600;
+
+            MediaDevice.DefaultAudioRenderDeviceChanged += OnDefaultAudioRenderDeviceChanged;
```

Comment:
Move this code to `ShellViewModel` `ctor` at the very end. Furthermore, this should be enclosed in a `if` conditional for checking device. As I understand, this event is only for PCs so there is no use registering it on mobile.

So this line would as a whole become when you put it at the end of ShellViewModel ctor:
```csharp
            if(ApiInformation.IsApiContractPresent("Windows.Phone.PhoneContract", 1))
                AudioRoutingManager.GetDefault().AudioEndpointChanged += Shell_AudioEndpointChanged;
            else
                MediaDevice.DefaultAudioRenderDeviceChanged += OnDefaultAudioRenderDeviceChanged;
```
The reason behind this is that ShellVM is called BEFORE SharedLogic. Alongside that, SharedLogic is initiated multiple times and does not maintain its state, so moving this to ShellVM to avoid multiple events being defined and registered would be good.